shift_detector/Detector.py

```diff
@@ -52,53 +52,6 @@ def add_checks(self, checks: List[Check]):
         self.checks_to_run += checks
         return self
 
-    def _needed_preprocessing(self, checks: List[Check]) -> Dict:
-        """
-        Find the needed preprocessings for each column type and
-        aggregate them.
-        :param checks: the checks the prepocessings will be aggregated from
-        :return: Dict
-        {
-            ColumnType: Set(Preprocessor, ...)
-            ...
-        }
-        """
-
-        def update_preprocessings(groups, checks):
-            for key, value in checks.needed_preprocessing().items():
-                groups[key].add(value)
-            return groups
-
-        type_to_needed_preprocessings = reduce(update_preprocessings, checks, defaultdict(set))
-        type_to_needed_preprocessings = dict(type_to_needed_preprocessings)
-        return type_to_needed_preprocessings
-
-    def _preprocess(self,
-                    column_type_to_columns: Dict,
-                    type_to_needed_preprocessings: Dict) -> Dict:
-        """
-        Execute the preprocessing.
-        :param column_type_to_columns: result of split_dataframes
-        :param type_to_needed_preprocessings: result of _needed_preprocessing
-        :return: Dict
-        {
-            ColumnType: {
-                Preprocessor: (df1_processed, df2_processed)
-                ...
-            }
-            ...
-        }
-        """
-        preprocessings = defaultdict(dict)
-
-        for column_type, needed_preprocessings in type_to_needed_preprocessings.items():
-            (first_df, second_df) = column_type_to_columns[column_type]
-            for needed_preprocessing in needed_preprocessings:
-                preprocessed = needed_preprocessing.process(first_df, second_df)
-                preprocessings[column_type][needed_preprocessing] = preprocessed
-
-        return preprocessings
-
     def _distribute_preprocessings(self, checks: List[Check], preprocessings: Dict):
```

Comment:
```suggestion
    @staticmethod
    def _distribute_preprocessings(self, checks: List[Check], preprocessings: Dict):
```

this method can be declared as static